### PR TITLE
Fix table cannot be recoginsed without empty line

### LINKED ISSUE #141: Table  is not rendered at first non-empty line
Hello cmark-gfm,

Thanks for you AMAZING Project

I noticed that the table  is not rendered at first non-empty line, such as

```markdown
abc
| abc | def |
| --- | --- |
| bar | baz |
```

it gives you the unexpected output

```
<p>abc
|abc|def|
|---|---|
|bar|baz|</p>
``` 

Is there any chance to be compatible with the table?  

Thanks a lot :)
JimCheung

### PR DESCRIPTION
**Pull request by jinhucheung**
fix #141

### CODE REVIEW

File: test/extensions.txt [hunk 1]
```diff
@@ -150,7 +150,7 @@ Hello!
 
 | _abc_ | セン |
 | ----- | ---- |
-| this row has a space at the end | | 
+| this row has a space at the end | |
```
Comment (kivikakk) [inline test/extensions.txt]:
```suggestion
| this row has a space at the end | | 
```

This was probably removed by accident.

Comment (jinhucheung) [reply test/extensions.txt]: This is "whitespace trimming" on vscode :heart:

File: extensions/table.c [hunk 2]
```diff
@@ -161,6 +175,25 @@ static table_row *row_from_string(cmark_syntax_extension *self,
...
+static void try_inserting_table_header_paragraph(cmark_parser *parser,
+                                                 cmark_node *parent_container,
+                                                 unsigned char *parent_string,
+                                                 int paragraph_offset) {
+  cmark_node *paragraph;
+  cmark_strbuf *paragraph_content;
+
+  paragraph = cmark_node_new_with_mem(CMARK_NODE_PARAGRAPH, parser->mem);
+
+  paragraph_content = unescape_pipes(parser->mem, parent_string, paragraph_offset);
```
Comment (jinhucheung) [inline extensions/table.c]: removed redundant type conversions 7f69c4e

**Comment by kivikakk** [comment]
I've pushed 15e924d from @philipturnbull, who found a memory leak while fuzzing this branch.

**Comment by kivikakk** [comment]
@jinhucheung Thank you for your PR!

**Pull request merged**